Dart/src/com/jetbrains/lang/dart/ide/refactoring/DartInlineHandler.java

```diff
@@ -0,0 +1,226 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jetbrains.lang.dart.ide.refactoring;
+
+import com.intellij.lang.Language;
+import com.intellij.lang.refactoring.InlineActionHandler;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiDocumentManager;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.refactoring.RefactoringBundle;
+import com.intellij.refactoring.inline.InlineOptionsDialog;
+import com.jetbrains.lang.dart.DartLanguage;
+import com.jetbrains.lang.dart.analyzer.DartAnalysisServerService;
+import com.jetbrains.lang.dart.analyzer.DartAnalysisServerService.PluginNavigationRegion;
+import com.jetbrains.lang.dart.analyzer.DartAnalysisServerService.PluginNavigationTarget;
+import com.jetbrains.lang.dart.assists.AssistUtils;
+import com.jetbrains.lang.dart.ide.actions.AbstractDartFileProcessingAction;
+import com.jetbrains.lang.dart.ide.refactoring.status.RefactoringStatus;
+import org.dartlang.analysis.server.protocol.ElementKind;
+import org.dartlang.analysis.server.protocol.SourceChange;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public class DartInlineHandler extends InlineActionHandler {
+  @Override
+  public boolean canInlineElement(PsiElement element) {
+    return false;
+  }
+
+  @Override
+  public boolean canInlineElementInEditor(PsiElement element, Editor editor) {
+    final InlineRefactoringContext context = findContext(editor);
+    if (context == null) return false;
+    final String kind = context.kind;
+    return ElementKind.LOCAL_VARIABLE.equals(kind) ||
+           ElementKind.METHOD.equals(kind) ||
+           ElementKind.FUNCTION.equals(kind) ||
+           ElementKind.GETTER.equals(kind) ||
+           ElementKind.SETTER.equals(kind);
+  }
+
+  @Override
+  public void inlineElement(@NotNull final Project project, @Nullable Editor editor, PsiElement element) {
+    final InlineRefactoringContext context = findContext(editor);
+    if (context == null) {
+      return;
+    }
+    // create refactoring
+    final ServerRefactoring refactoring;
+    if (ElementKind.LOCAL_VARIABLE.equals(context.kind)) {
+      refactoring = new ServerInlineLocalRefactoring(context.filePath, context.offset, 0);
+    } else {
+      refactoring = new ServerInlineMethodRefactoring(context.filePath, context.offset, 0);
```

Comment:
A mess with system dependent/independent slashes again. Somewhere we convert paths inside `DartAnalysisServerService` methods, somewhere - outside. I had to tweak to make it work on Windows.

Suggestion: work with system independent paths everywhere outside `DartAnalysisServerService` methods. Convert paths to system dependent only locally in the methods that call server. Convert paths back to system independent as soon as result is received. Make sure no fields contain system dependent slashes and no parameters expect to receive system dependent ones and no method returns system dependent path.